### PR TITLE
LSTM reg

### PR DESCRIPTION
**Pull request by Ashutosh-Adhikari**
Add LSTM reg

### CODE REVIEW

File: lib/models/sm_lstm/model.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        else:
+            self.fc1 = nn.Linear(2 * config.hidden_dim, config.target_class)
+
+        if config.beta_ema>0:
+            self.avg_param = deepcopy(list(p.data for p in self.parameters()))
+            if torch.cuda.is_available():
+                self.avg_param = [a.cuda() for a in self.avg_param]
+            self.steps_ema = 0.
+
+    def forward(self, query, text):
```
Comment (achyudh) [inline lib/models/sm_lstm/model.py]: Shouldn't we be calling this methods with the sequence lengths? The lengths are passed in from the trainer as a tuple. See `torchtext_trainer.py` for the exact function call.

File: lib/models/sm_lstm/encoder.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            x = torch.nn.utils.rnn.pack_padded_sequence(x, lengths, batch_first=True)
+        rnn_outs, _ = self.lstm(x)
+        rnn_outs_temp = rnn_outs
+
+        if lengths is not None:
+            rnn_outs,_ = torch.nn.utils.rnn.pad_packed_sequence(rnn_outs, batch_first=True)
+            rnn_outs_temp, _ = torch.nn.utils.rnn.pad_packed_sequence(rnn_outs_temp, batch_first=True)
+
+        x = F.relu(torch.transpose(rnn_outs_temp, 1, 2))
+        x = F.max_pool1d(x, x.size(2)).squeeze(2)
```
Comment (achyudh) [inline lib/models/sm_lstm/encoder.py]: `return x`?

File: lib/models/sm_lstm/encoder.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        '''x = self.dropout(x)
```
Comment (achyudh) [inline lib/models/sm_lstm/encoder.py]: I'd prefer if we remove this commented out block of code

**Comment by achyudh** [comment]
@Ashutosh-Adhikari I am closing this in favour of a more flushed out version with all the required files and dependencies: https://github.com/castorini/errol/pull/7

**Comment by Ashutosh-Adhikari** [comment]
@achyudh why are we closing this issue? What do you mean by more flushed out version? I addressed the comments.

**Comment by achyudh** [comment]
You addressed one of the three comments. But flushed out I meant having a driver method, arguments and integrating it with the existing datasets.

**Comment by Ashutosh-Adhikari** [comment]
oh i saw only one of the comments lol. My bad.

**Pull request closed without merging**